rpc/src/main/scala/org/bitcoins/rpc/client/RpcClient.scala

```diff
@@ -0,0 +1,1114 @@
+
+  def listUnspent(
+      addresses: Vector[BitcoinAddress]): Future[Vector[UnspentOutput]] =
+    listUnspent(addresses = addresses)
+
+  def listUnspent(
+      minConfirmations: Int,
+      maxConfirmations: Int,
+      addresses: Vector[BitcoinAddress]): Future[Vector[UnspentOutput]] =
+    listUnspent(minConfirmations, maxConfirmations, Some(addresses))
+
+  def listWallets: Future[Vector[String]] = {
+    bitcoindCall[Vector[String]]("listwallets")
+  }
+
+  def lockUnspent(
+      unlock: Boolean,
+      outputs: Vector[RpcOpts.LockUnspentOutputParameter]): Future[Boolean] = {
+    bitcoindCall[Boolean]("lockunspent",
+                          List(JsBoolean(unlock), Json.toJson(outputs)))
+  }
+
+  private def logging(
+      include: Option[Vector[String]],
+      exclude: Option[Vector[String]]): Future[Map[String, Int]] = {
+    val params =
+      if (include.isEmpty && exclude.isEmpty) {
+        List.empty
+      } else if (include.isEmpty) {
+        List(JsArray.empty, Json.toJson(exclude.get))
+      } else if (exclude.isEmpty) {
+        List(Json.toJson(include.get), JsArray.empty)
+      } else {
+        List(Json.toJson(include.get), Json.toJson(exclude.get))
+      }
+    bitcoindCall[Map[String, Int]]("logging", params)
+  }
+  def logging: Future[Map[String, Int]] = logging(None, None)
+
+  def logging(
+      include: Vector[String] = Vector(),
+      exclude: Vector[String] = Vector()): Future[Map[String, Int]] = {
+    val inc = if (include.nonEmpty) Some(include) else None
+    val exc = if (exclude.nonEmpty) Some(exclude) else None
+    logging(inc, exc)
+  }
+
+  def move(
+      fromAccount: String,
+      toAccount: String,
+      amount: Bitcoins,
+      comment: String = ""): Future[Boolean] = {
+    bitcoindCall[Boolean]("move",
+                          List(JsString(fromAccount),
+                               JsString(toAccount),
+                               JsNumber(amount.toBigDecimal),
+                               JsNumber(6),
+                               JsString(comment)))
+  }
+
+  def ping(): Future[Unit] = {
+    bitcoindCall[Unit]("ping")
+  }
+
+  def preciousBlock(headerHash: DoubleSha256Digest): Future[Unit] = {
+    bitcoindCall[Unit]("preciousblock", List(JsString(headerHash.hex)))
+  }
+
+  def prioritiseTransaction(
+      txid: DoubleSha256Digest,
+      feeDelta: Satoshis): Future[Boolean] = {
+    bitcoindCall[Boolean](
+      "prioritisetransaction",
+      List(JsString(txid.hex), JsNumber(0), JsNumber(feeDelta.toLong)))
+  }
+
+  def pruneBlockChain(height: Int): Future[Int] = {
+    bitcoindCall[Int]("pruneblockchain", List(JsNumber(height)))
+  }
+
+  def removePrunedFunds(txid: DoubleSha256Digest): Future[Unit] = {
+    bitcoindCall[Unit]("removeprunedfunds", List(JsString(txid.hex)))
+  }
+
+  private def rescanBlockChain(
+      start: Option[Int],
+      stop: Option[Int]): Future[RescanBlockChainResult] = {
+    val params =
+      if (start.isEmpty) {
```

Comment:
Is there a missing case here? Where we start scanning from the genesis block, but stop at a given block? Perhaps pattern match on this so you get an exhaustiveness check?

Reply:
This actually isn't possible since rescanblockchain does not take a map as input but rather it takes 0, 1, or 2 integers and so it always interprets a single Int as the start parameter